Fixes Vat-Grown Naming

Sweedle opened the pull request:
<!-- 
Do not forget to add a changelog when you have made admin/player facing changes that can alter gameplay.
Examples which require a changelog entry include:
* Adding/removing objects that players may interact with, or the way they function.
* Adding/removing/altering admin tools.
* Changing the map.

Examples were changelog entries are optional/not typically required:
* Cosmetic changes such as descriptions, sound effects, etc.
* Optimizations and other changes to underlying systems which do not affect gameplay.
* Minor bug fixes.

You'll find a README and example file in .\html\changelogs\ for further instructions.

You can also find a template for adding your changelog directly to the PR description here: https://github.com/Baystation12/Baystation12/wiki/Automatic-changelog-generation
-->
I fixed this thing. Vat-Growns will no longer spawn with just a first name. Existing naming schemes work as before.

Fixes #24938

Sweedle committed:
Changes Vat-Grown Naming

MistakeNot4892 commented on code/modules/culture_descriptor/culture/cultures_human.dm:
```diff
@@ -146,13 +146,14 @@
 	#undef NUM
-	#undef NAME
\ No newline at end of file
+	#undef NAME
```
You never undef `FIRST`.

Sweedle replied:
Whoops, resolved now.

MistakeNot4892 reviewed (changes requested):
As written, unless I am misunderstanding, this commit changes nothing whatsoever about the naming behavior. You've renamed `NAME` to `FIRST` and added a new `NAME` define that you then don't actually use.

Sweedle committed:
Undefs FIRST

Sweedle commented:
> As written, unless I am misunderstanding, this commit changes nothing whatsoever about the naming behavior. You've renamed `NAME` to `FIRST` and added a new `NAME` define that you then don't actually use.

Correct me if I'm wrong -- I probably am -- `NAME` is used on the if(1) roll.

Previously, `NAME` was just a first name. I changed `NAME` to be First Last, meaning the if(1) roll was fine with the changed definition -- it wasn't changed, unlike the others.

`FIRST` was created to be used in other places where `NAME` had been, to avoid names like "Steven James-123"

		**if(1) return NAME**
		if(2) return "[LTR][LTR]-[FIRST]"
		if(3) return "[FIRST]-[NUM][NUM][NUM]"
		if(4) return "[LTR][LTR]-[NUM][NUM][NUM]"

MistakeNot4892 commented:
Missed the first case, my mistake. Let's call it 'change blindness' like snow blindness.
